src/webhooks/sentry-options/sentry-options.ts

```diff
@@ -205,23 +121,84 @@ export async function messageSlack(message: SentryOptionsResponse) {
 }
 
 async function sendMessage(blocks) {
-  for (const splitBlock of blocks) {
-    try {
-      await bolt.client.chat.postMessage({
-        channel: FEED_OPTIONS_AUTOMATOR_CHANNEL_ID,
-        blocks: splitBlock,
-        text: '',
-        unfurl_links: false,
-      });
-    } catch (err) {
-      Sentry.captureException(err);
-    }
+  try {
+    await bolt.client.chat.postMessage({
+      channel: FEED_OPTIONS_AUTOMATOR_CHANNEL_ID,
+      blocks: blocks,
+      text: '',
+      unfurl_links: false,
+    });
+  } catch (err) {
+    Sentry.setContext('block:', { blocks });
+    Sentry.captureException(err);
+  }
+}
+
+function generateBlock(option_type: string, options: any[]): KnownBlock[] {
+  /**
+   * This function generates a list of KnownBlocks, a type of SlackBlock.
+   * If the given option_type does not fit into the formatterMap, report it as a sentry error.
+   * This function also builds blocks around the options in batches of MAX_BLOCK_SIZE. Each
+   * sectionBlock has a block limit.
+   *
+   */
+
+  if (options.length === 0) {
+    return [];
+  }
+
+  const formatterMap: { [key: string]: OptionFormatter } = {
```

Comment:
nit: Should this live upstream somewhere? Given that `generateBlock` is called multiple times, meaning this formatterMap will be recreated on each call

Reply:
I moved this (and the relevant functions) to be better scoped.